Workshop: Node First Girls Express (#169)
CodeYourFuture/CYF-Workshops: express-setup/readme.md

To use `fs`, require it at the top of your server file:

```js
const fs = require("fs");
```

The method we need to write to your hard drive is `fs.writeFile`.

```js
fs.writeFile("path/to/file", yourData, (error) => {
  // do something
});
```

- Argument 1: `"path/to/file"` the location of the file you want to write to
- Argument 2: `yourData` the data you want to write
- Argument 3: `(error) =>` the callback function

The 'path/to/file' will be replaced with the actual path to the file you want to write to. If it doesn't exist, `fs.writeFile` cleverly creates one for you. But we already have `posts.json`, so not to worry.

### Reading from your hard drive

To read data that's already there, you would use `fs.readFile`. The way to use `fs.readFile` is very similar to `fs.writeFile`:

```js
fs.readFile("path/to/file", (error, file) => {
  // do something
});
```

- Argument 1: the location of the file you want to read from
- Argument 2: the callback function

> You'll notice that `fs.readFile`'s callback function takes a second argument. That argument is the file you're _reading_.

Reviewer: ```suggestion
```

Let's read the data from the `posts.json` file. Make sure you've `require`d the `fs` core Node module at the top of your server file somewhere.

Add this code to your server (put it anywhere after the `require`s for now):

```js
fs.readFile(__dirname + "/data/posts.json", (error, file) => {
  console.log(file);
});
```

(`__dirname` is a Node global object that gives you a path to your current working directory. It's handy to avoid writing the whole path out in full.)

If you restart the server, you'll probably see something like this:

```bash
<Buffer 7b 0a 20 20 20 20 22 31 34 36 37 33 39 30 33 35 36 32 39 31 22 3a 20 22 54 68 69 73 20 69 73 20 6d 79 20 76 65 72 79 20 66 69 72 73 74 20 62 6c 6f 67 ... >
```

This is actually the contents of your `posts.json` file, but in a format called a **buffer**. To make it a bit more human-readable,console.log the file to a string, like this:

```js
console.log(file.toString());
```

`file` is in JSON format right now. To access the blog post message inside `file`, we need to parse it from JSON back to a JavaScipt object.

> Add this next bit of code to your `fs.readFile`'s callback function:

Reviewer: ```suggestion
Add this next bit of code to the `.then` callback for `fs.readFile`'s:
```
